### PR TITLE
Implemented backend and frontend for student usage

### PR DESCRIPTION
**Pull request by aaronkang7**
### Summary 

This pull request implements backend endpoints and frontend elements for usage element for the students' table. 

Screen Shot 2021-04-21 at 9 12 44 PM

### Breaking Changes  

<!-- - Database schema change (anything that changes DynamoDB document structure)
<!-- - Other change that could cause problems (Detailed in notes)

### CODE REVIEW

File: server/router/rider.ts [hunk 1]
```diff
@@ -80,6 +80,26 @@ router.get('/:id/currentride', validateUser('Rider'), (req, res) => {
...
+    const conditionNoShow = isRider.group(isNoshow);
+    const isCompleted = new Condition('status').eq(Status.COMPLETED);
+    const conditionRides = isRider.group(isCompleted);
+    db.scan(res, Ride, conditionNoShow, (data: RideType[]) => {
+      noShowCount = data.length;
+      db.scan(res, Ride, conditionRides, (data_: RiderType[]) => {
+        studentRides = data_.length;
+        res.send({ studentRides, noShowCount });
+      });
+    });
```
Comment (mg2239) [inline server/router/rider.ts:99]: to reduce database reads, you could use the isRider condition, then do two .filter() calls, one for no show, one for completed, on the data sent from the callback

File: frontend/src/components/UserTables/StudentsTable.tsx [hunk 2]
```diff
@@ -1,13 +1,38 @@
...
+        <span>
+          <span className={styles.usageContainer}>
+            <span className={cn(styles.ridesCount, styles.usageTag)}></span>
+            {usage.studentRides} Rides
+          </span>
+          <span className={styles.usageContainer}>
+            <span className={cn(styles.noShow, styles.usageTag)}></span>
+            {usage.noShowCount} No Shows
+          </span>
+        </span>,
```
Comment (mg2239) [inline frontend/src/components/UserTables/StudentsTable.tsx:32]: can you make them on the same line like in the designs? should be just a flexbox w/ some positioning

**Review by mg2239 — CHANGES_REQUESTED** [review]
nice work! good job figuring out the server stuff, just a couple small changes

File: frontend/src/components/UserTables/StudentsTable.tsx [hunk 3]
```diff
@@ -1,13 +1,38 @@
...
+            <span className={cn(styles.ridesCount, styles.usageTag)}></span>
+            {usage.studentRides} Rides
+          </span>
+          <span className={styles.usageContainer}>
+            <span className={cn(styles.noShow, styles.usageTag)}></span>
+            {usage.noShowCount} No Shows
+          </span>
+        </div>,
+    };
+  };
```
Comment (mg2239) [inline frontend/src/components/UserTables/StudentsTable.tsx:34]: just found a bug: usage is the state for the whole table, not just an individual rider. so every rider has the same usage at the moment. we'd either need to make a request each per user (which would be expensive) and figure out a way to keep track of that in state, or create another endpoint to get the usage of all riders in the table. if you have time id like you to try the second option, if not i can try to do it within the next couple days

**Review by mg2239 — CHANGES_REQUESTED** [review]

File: frontend/src/components/UserTables/StudentsTable.tsx (hunk 3 above)
Comment (aaronkang7) [reply frontend/src/components/UserTables/StudentsTable.tsx:34]: i may have some time to work on it this weekend but I don't think I'll be able to finish it by the work session :(

**Review by mg2239 — APPROVED** [review]
nice work!

**Pull request merged**